app/services/search/query_builders/reaction.rb

```diff
@@ -64,7 +64,13 @@ def terms_keys
         TERM_KEYS.map do |term_key, search_key|
           next unless @params.key? term_key
 
-          { terms: { search_key => Array.wrap(@params[term_key]) } }
+          values = Array.wrap(@params[term_key])
+
+          if params[:tag_boolean_mode] == "all"
```

Comment:
Ideally, I would like to add this to all search query builders but I will save that for a future refactor. Then in the view we can allow users to toggle between AND and OR options for searching multiple items.